src/Button.php

```diff
@@ -4,95 +4,378 @@
+     * @return self A new instance with the specified 'data-bs-toggle' value.
+     */
+    public function dataBsToggle(string $value): self
+    {
+        $new = clone $this;
+        $new->attributes['data-bs-toggle'] = $value;
+
+        return $new;
+    }
+
+    /**
+     * Sets the button to be disabled.
+     *
+     * @param bool $value Whether the button should be disabled.
+     *
+     * @return self A new instance with the button disabled.
+     */
+    public function disabled(bool $value = true): self
+    {
+        $new = clone $this;
+        $new->disabled = $value;
+
+        return $new;
+    }
+
+    /**
+     * Sets the ID of the button component.
+     *
+     * @param bool|string $value The ID of the button component. If `true`, an ID will be generated automatically.
+     *
+     * @return self A new instance with the specified ID.
+     */
+    public function id(bool|string $value): self
+    {
+        $new = clone $this;
+        $new->id = $value;
+
+        return $new;
+    }
+
+    /**
+     * The button label.
+     *
+     * @param string $value The label to display on the button.
+     * @param bool $encode Whether the label value should be HTML-encoded. Use this when rendering user-generated
+     * content to prevent XSS attacks.
+     *
+     * @return self A new instance with the specified label value.
+     */
+    public function label(string|Stringable $value, bool $encode = true): self
+    {
+        if ($encode) {
+            $value = Html::encode($value);
+        }
+
         $new = clone $this;
         $new->label = $value;
 
         return $new;
     }
 
     /**
-     * The HTML attributes for the widget container tag. The following special options are recognized.
+     * Sets the button size to be large.
+     *
+     * @return self A new instance with the button as a large button.
+     */
+    public function largeSize(): self
+    {
+        $new = clone $this;
+        $new->cssClass['size'] = 'btn-lg';
+
+        return $new;
+    }
+
+    /**
+     * Whether the button should be a link.
+     *
+     * @param string|null $url The URL of the link.
+     *
+     * @return self A new instance with the button as a link.
+     */
+    public function link(string|null $url = null): self
+    {
+        $new = clone $this;
+        $new->tag = A::tag()->url($url);
+
+        return $new;
+    }
+
+    /**
+     * Sets the button size to be normal.
+     *
+     * @return self A new instance with the button as a normal button.
+     */
+    public function normalSize(): self
+    {
+        $new = clone $this;
+        $new->cssClass['size'] = null;
+
+        return $new;
+    }
+
+    /**
+     * Whether the button should be a reset button.
+     *
+     * @param string|null $value The content of the button. For default, it is 'Reset'.
+     *
+     * @return self A new instance with the button as a reset button.
+     */
+    public function reset(string|null $value = 'Reset'): self
```

Comment:
Why create another static method, if we already have the factory in `::widget` where i can pass the arguments without problems?

Reply:
Yes, i agree that static methods are shorter and nicer, but they have some considerations.

- Changing them to static methods since they are immutable, we lose the ability to change the buttons dynamically according to the state of the form.

Reply:
Correct. That is always a tradeoff.

Reply:
done.